### PR TITLE
Add x87 instructions intended for use with pre-encoded registers

### PR DESCRIPTION
**Pull request by BradleyWood**
Add x87 instructions for pre-encoding registers to lay groundwork for removal of x87. See #946

Enable SSE3 feature check which will be required for FP calculations

### CODE REVIEW

File: compiler/x/codegen/X86Ops.ins [hunk 1]
```diff
@@ -1183,6 +1187,10 @@ INSTRUCTION(FLDRegReg, fld,
             BINARY(VEX_L___, VEX_vNONE, PREFIX___, REX__, ESCAPE_____, 0xd9, X87_________________(0xc0)),
             PROPERTY0(IA32OpProp_ModifiesTarget | IA32OpProp_SingleFP),
             PROPERTY1(IA32OpProp1_TargetRegIsImplicit)),
+INSTRUCTION(FLDDUP, fld,
+            BINARY(VEX_L___, VEX_vNONE, PREFIX___, REX__, ESCAPE_____, 0xd9, X87_________________(0xc0)),
+            PROPERTY0(IA32OpProp_ModifiesTarget | IA32OpProp_SingleFP),
```
Comment (0xdaryl) [inline compiler/x/codegen/X86Ops.ins:1192]: I don't think the `IA32OpProp_SingleFP` property makes sense unless you also define a `DLDDUP` instruction, which I don't think you need to.

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Please fix the length of the commit title for b8446c9981d4d524ac3c0410b3568e1e623d55b0 so that it doesn't wrap with ellipses.

File: compiler/x/codegen/X86Ops.ins (hunk 1 above)
Comment (BradleyWood) [reply compiler/x/codegen/X86Ops.ins]: Fixed in [force-push](https://github.com/eclipse/omr/compare/1d77ffe76e2dc9410d223c0a72faa1f4fed3dc96..dcfb579327cf5c1dc2c7b912198553b070d81cbd)

**Comment by 0xdaryl** [comment]
Jenkins build xlinux,xosx,xwindows

**Comment by 0xdaryl** [comment]
Jenkins build linux_x86

**Comment by 0xdaryl** [comment]
Jenkins build xlinux32

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**